From 07655c7c8eac00fd7454ee641b53faeb2f954fe6 Mon Sep 17 00:00:00 2001
From: Steven Moreland <smoreland@google.com>
Date: Wed, 18 Nov 2020 01:16:39 +0000
Subject: [PATCH] セキュリティパッチ 2021-03-01 ANDROID-172655291

[porting]

hidl_test_java: reflect new overread check in art

This was passing before because of some other checks which caught the
mismatch resulting from an overread.

Ignore-AOSP-First: security

Bug: 172655291
Test: atest hidl_test_java
Merged-In: Ie3766d20597ed9714ab3c3beead8af01226f58ba

CVE-2021-0394

Change-Id: Ie3766d20597ed9714ab3c3beead8af01226f58ba
(cherry picked from commit aeae74893c6973959fe58b6e8e045fc4be2b8d32)
(cherry picked from commit f9a784013d8b6d519e66c2bee6384ad8a713ac25)
---
 test/java_test/hidl_test_java_native.cpp | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/test/java_test/hidl_test_java_native.cpp b/test/java_test/hidl_test_java_native.cpp
index fff777f..4c479f2 100644
--- a/test/java_test/hidl_test_java_native.cpp
+++ b/test/java_test/hidl_test_java_native.cpp
@@ -301,13 +301,11 @@ TEST_F(HidlTest, SomeOtherBaseMethodInvalidString) {
         }
     };
 
-    auto ret = baz->someOtherBaseMethod(foo, [&](const auto&) {
-        ADD_FAILURE() << "Should not accept invalid UTF-8 String";
+    auto ret = baz->someOtherBaseMethod(foo, [](const IBase::Foo& ret) {
+        EXPECT_EQ(ret.y.s, "?");  // :)
     });
 
-    EXPECT_FALSE(ret.isOk());
-
-    EXPECT_OK(baz->ping());
+    EXPECT_TRUE(ret.isOk());
 }
 
 TEST_F(HidlTest, BazSomeMethodWithFooArraysTest) {
-- 
2.30.0.280.ga3ce27912f-goog

